### ISSUE TITLE
Create surveyor sheet mockup

### ISSUE THREAD

**Issue by dbl3raf** [bird-team/brisbane-bird-atlas#65]

**Comment by dbl3raf** [comment]
Here's the draft mockup. I'm expecting this to go through another iteration, but this is the basic idea. At this stage I'm wondering how tractable it would be to automatically produce an image like the one used in this for each of the grid cells? At present I'm favouring producing the survey sheets manually - a couple of blokes are going to help - but if the images could be produced somehow via an automated process that would really help. If it's not easy to set up something that I could then inherit and use, it's probably best that I just push ahead and make these myself from ArcGIS - it really wouldn't be too onerous.

As you can see, I've labelled each location in the eBird data within the square, so the user can see which spots have already been visited.

[Atlas of the Birds of Brisbane Grid Square Summary Sheet_02.docx](https://github.com/bird-team/brisbane-bird-atlas/files/2401366/Atlas.of.the.Birds.of.Brisbane.Grid.Square.Summary.Sheet_02.docx)

**Comment by jeffreyhanson** [comment]
Yeah that looks awesome! I could definitely automate creating the maps. Although if you wanted to automate the entire thing, then we could look at creating a "companion" bookdown repository which has all the survey sheets - but this would probably take a lot longer.

**Comment by jeffreyhanson** [comment]
@dbl3raf, can you please confirm if you're happy with the grid cells present in the _data/grid/grid.shp_ file? If so, I'll start creating all the images for them.

**Comment by jeffreyhanson** [comment]
I'm just making the maps now. On the mockup, the map has purple points with names - did you create these manually or are they just part of the background map? If the former, is there a database which contains the lon/lats and names of these points? If so, is this from the locality data in the eBird records or somewehre else?

**Comment by dbl3raf** [comment]
![map](https://user-images.githubusercontent.com/23551540/46118049-f98c7b00-c247-11e8-9a29-783fd625583a.JPG)
keen for styling like the attached if possible. The coordinates and labels are directly from the eBird data, with a (P) appended for personal locations and (H) for hotspots. I faced a few challenges around label placement, especially where the density of locations in a grid square is high, or the names are long. If this becomes too big an issue, we could restrict the locations shown to just hotspots.

**Comment by jeffreyhanson** [comment]
Ok - can do. Just to double check, did you want maps for all sampling grids - or just terrestrial ones? There are a total of 10k sampling grids including the marine grids.

**Comment by dbl3raf** [comment]
just terrestrial. Any chance I can quickly review some examples first, or is it just as quick to build the whole lot, and then circle back to make any small adjustments?

**Comment by jeffreyhanson** [comment]
Ok - yeah, I'll send you five to review before making the whole lot.

**Comment by jeffreyhanson** [comment]
I can't find a column in the eBird record data indicating if a locality is a "hotspot" or a "personal" location. I've copied in the column names below - can you please let me know which column to use?

```
 [1] "GLOBAL UNIQUE IDENTIFIER"     "LAST EDITED DATE"            
 [3] "TAXONOMIC ORDER"              "CATEGORY"                    
 [5] "COMMON NAME"                  "SCIENTIFIC NAME"             
 [7] "SUBSPECIES COMMON NAME"       "SUBSPECIES SCIENTIFIC NAME"  
 [9] "OBSERVATION COUNT"            "BREEDING BIRD ATLAS CODE"    
[11] "BREEDING BIRD ATLAS CATEGORY" "AGE/SEX"                     
[13] "COUNTRY"                      "COUNTRY CODE"                
[15] "STATE"                        "STATE CODE"                  
[17] "COUNTY"                       "COUNTY CODE"                 
[19] "IBA CODE"                     "BCR CODE"                    
[21] "USFWS CODE"                   "ATLAS BLOCK"                 
[23] "LOCALITY"                     "LOCALITY ID"                 
[25] "LOCALITY TYPE"                "LATITUDE"                    
[27] "LONGITUDE"                    "OBSERVATION DATE"            
[29] "TIME OBSERVATIONS STARTED"    "OBSERVER ID"                 
[31] "SAMPLING EVENT IDENTIFIER"    "PROTOCOL TYPE"               
[33] "PROTOCOL CODE"                "PROJECT CODE"                
[35] "DURATION MINUTES"             "EFFORT DISTANCE KM"          
[37] "EFFORT AREA HA"               "NUMBER OBSERVERS"            
[39] "ALL SPECIES REPORTED"         "GROUP IDENTIFIER"            
[41] "HAS MEDIA"                    "APPROVED"                    
[43] "REVIEWED"                     "REASON"                      
[45] "TRIP COMMENTS"                "SPECIES COMMENTS"            
[47] "V47"
```

**Comment by dbl3raf** [comment]
"LOCALITY TYPE"

**Comment by jeffreyhanson** [comment]
Great - thanks

**Comment by jeffreyhanson** [comment]
Just as a heads up, there's 564 terrestrial grid cells (defined as cells which partially overlap with land)

**Comment by dbl3raf** [comment]
brilliant - thanks, it's a good number.

**Comment by jeffreyhanson** [comment]
Here's some examples. Note that I've only included locality names for localities that fall inside the primary grid cell (i.e. the grid cell for which the image was generated). Some of the examples below do not include any sampling localities since they are along the coast. To avoid overlapping text, I am using the [ggrepel R package](https://cran.r-project.org/web/packages/ggrepel/vignettes/ggrepel.html) to automatically place labels in a manner that avoids overlaps (but note that due to this it can result in some arrangements that appear suboptimal). Also, if desired, I could change the text color to black and put a white box underneath the text?

![grid-9009](https://user-images.githubusercontent.com/3610005/46124253-5434cf80-c266-11e8-9907-7e5e55456749.png)
![grid-9101](https://user-images.githubusercontent.com/3610005/46124254-54cd6600-c266-11e8-877a-4674cc20e091.png)
![grid-9102](https://user-images.githubusercontent.com/3610005/46124255-54cd6600-c266-11e8-977b-bfd432b1b007.png)
![grid-9103](https://user-images.githubusercontent.com/3610005/46124256-5565fc80-c266-11e8-920f-da612fc70568.png)
![grid-9104](https://user-images.githubusercontent.com/3610005/46124257-5565fc80-c266-11e8-8c8a-74ff6de07047.png)

**Comment by jeffreyhanson** [comment]
Also, just as a heads up, I'm storing the code for producing these maps in a different repo: [bird-team/sampling-grid-maps](https://github.com/bird-team/sampling-grid-maps)

**Comment by jeffreyhanson** [comment]
For comparison, here's an example with a black transparent background:

![grid-9104](https://user-images.githubusercontent.com/3610005/46125553-05d6ff00-c26d-11e8-982a-3a18decceb69.png)

**Comment by dbl3raf** [comment]
looking brilliant Jeff - some tweaks:
(i) white grid squares - red doesn't work for colour blind people and won't print well in B&W
(ii) much smaller font size for labels, about half current size
(iii) definitely keep the black semitransparent background so that the labels are emphasized, but remove the white border around the label background - this will reduce clutter and increase ability to see the map underneath

**Comment by jeffreyhanson** [comment]
Ok - I'll see what I can do.

**Comment by jeffreyhanson** [comment]
How's this?

![grid-9104](https://user-images.githubusercontent.com/3610005/46181301-3f107d00-c308-11e8-872a-e3f57ffcca1c.png)

**Comment by jeffreyhanson** [comment]
If you wanted to reduce the amount of text in each label, we could color code the labels and remove the 
"(H)" and "(P)" (e.g. P=white and H=yellow, or does eBird have a color coding scheme)?

**Comment by dbl3raf** [comment]
looking great! Couple of last things:

1. Could we shrink down the semitransparent background so it hugs the label more tightly, exposing more of the underlying map

2. Could we use different symbols, e.g. triangle for hotspot and circle for personal location? Colour coding risky I think because any printing would mostly happen in B&W

**Comment by jeffreyhanson** [comment]
Ok sure - give me a moment

**Comment by jeffreyhanson** [comment]
Should the image include a legend specifying what triangle=hotspot and circle=personal location? Or will this be in a caption in the sheet?

**Comment by dbl3raf** [comment]
Legend will be in a caption in the sheet

**Comment by jeffreyhanson** [comment]
Ok, how's this?

![grid-9104](https://user-images.githubusercontent.com/3610005/46182746-02954f00-c311-11e8-851c-40847c81cb7a.png)

**Comment by dbl3raf** [comment]
brilliant Jeff, many thanks

**Comment by jeffreyhanson** [comment]
No worries - I've made all the maps and uploaded them here: https://github.com/bird-team/sampling-grid-maps/releases (warning total size is 1.5GB).

**Comment by jeffreyhanson** [comment]
I'll close this issue now - but feel free to reopen it if you'd like any more changes to the maps.

**Comment by dbl3raf** [comment]
reopening this issue as we have now made a full draft of a survey sheet. The question for @jeffreyhanson is whether it is possible to produce something like this programmatically with only a modest amount of development time. I have a fairly streamlined workflow to create these, so doing it manually is not out of the question. Our initial chats have suggested that a programmatic solution here might need rather more time for development than @jeffreyhanson has available. Any thoughts?
[Atlas of the Birds of Brisbane Grid Square Summary Sheet_03.docx](https://github.com/bird-team/brisbane-bird-atlas/files/2592689/Atlas.of.the.Birds.of.Brisbane.Grid.Square.Summary.Sheet_03.docx)

**Comment by jeffreyhanson** [comment]
Yeah, it looks like I could come up with a programmatic solution, but it might take a few weekends or so to finish. It looks like most of it could be automated, but it has a section of text below the image that likely needs to be tweaked which makes things a bit tricky. If that's the only manually written text that is going to differ between surveyor sheets, we could have all that in a spreadsheet (where different rows correspond to different grid cells, and there is a "text" column that contains the description). The other trick will be creating the long table that's folded across the page, but it looks like this is achievable in latex (https://tex.stackexchange.com/questions/322034/folding-long-narrow-tables-within-a-page).

**Comment by Louis-Backstrom** [comment]
This looks great - and I think is going to be a huge part of the completed atlas. Only thing I've got to add right now - perhaps instead of triangles and circles for the two types of locations why couldn't we just have a P and a H (either enclosed in a circle/square or just on their own) as the location icons? I feel that would make more sense and be easier to remember/be reminded of rather than having to remember circle means hotspot or whatever?

**Comment by dbl3raf** [comment]
mainly for @Louis-Backstrom's interest - here is the Kedron survey sheet (only current to May 2018)
[Atlas of the Birds of Brisbane Grid Square Summary Sheet_04_Kedron.docx](https://github.com/bird-team/brisbane-bird-atlas/files/2592764/Atlas.of.the.Birds.of.Brisbane.Grid.Square.Summary.Sheet_04_Kedron.docx)

**Comment by jeffreyhanson** [comment]
Just a few questions regarding the mockup:

1. How is "total minutes" and "total km" assessed in the first table?

2. Does the second table show data for records or only true checklists?

**Comment by jeffreyhanson** [comment]
Also, should checklists and records be excluded before certain years? I.e. should records be excluded based on year data in the altas_list.xslx?

**Comment by dbl3raf** [comment]
1. Total minutes and total km is calculated by adding up the durations and distances respectively of all complete checklists (excluding those before “checklist” year in atlas_list.xlsx)

2. The Observer league table is for all records (excluding those before “records” year in atlas_list.xlsx)

3. The giant seasonal table is for all records (excluding those before “records” year in atlas_list.xlsx)

**Comment by jeffreyhanson** [comment]
Ok - just to double check - the checklist and record years are on a per species basis. This means that certain checklists which contain only certain species might get omitted from the distance/durations sums. For example, if a checklist was made in 1990 that contains only birds that have checklist years >= 2005, then all of the entries for this checklist will get omitted and it won't contribute to the checklist durations and distances. But if this same checklist had an additional bird that had a checklist year >= 1980, then this entry would remain, and the checklist would count towards the durations and distances. Is this behaviour desirable?

**Comment by jeffreyhanson** [comment]
Also, I really like this idea of a leader board and wonder if it's worth having a dedicated section in the atlas that has a Brisbane-wide leader board? This leader board could acknowledge people who have "contributed the most overall", but also people who have "contributed the most by checklisting in undersampled areas" since the previous atlas update?

**Comment by jeffreyhanson** [comment]
The observers' names don't seem to be in the raw checklist data, instead we just have their "OBSERVER ID". What is the best way to determine an observer's name from their ID?

**Comment by dbl3raf** [comment]
Yes that's the right behaviour in terms of checklists / records.

**Comment by dbl3raf** [comment]
ah yes, the observers. Well, eBird removed all observers names from the downloads because of new European privacy laws. Although we could use an old download to create a lookup table to get between observer ID and name, that would not really be acceptable use of data in a public repo. I think actually we'll have to remove the league table from the sheet, if we create it programatically, as I can't see any way around it.

**Comment by jeffreyhanson** [comment]
I have an idea. For a given observer id, we could find a checklist completed by the observer, extract the checklist id, download the web page for that checklist (e.g. https://ebird.org/view/checklist/S42627306), and extract the person's name from the web page? This could be done programmatically using web scraping - do you know if eBird's terms of service forbid this?

**Comment by Louis-Backstrom** [comment]
I can't speak for whether eBird's TOS forbid that, but I'm not sure how well that would work with shared checklists with 2+ observers listed? I'm not entirely sure how eBird handles checklist IDs with multiple observers but it's probably worth keeping in mind.

**Comment by jeffreyhanson** [comment]
Yeah thanks for raising that point. Maybe we could add in a step that extracts only checklist ids for non-shared checklists?

**Comment by dbl3raf** [comment]
A checklist shared, for example, between two observers results in two separate checklists being stored in the eBird database. Each will have a unique sampling event ID, and only a single observer ID. Shared checklists are linked with the Group Identifier. it is done this way to allow individuals to edit "their own" checklist if, for example, they didn't see a species that someone else saw.

**Comment by jeffreyhanson** [comment]
Ah ok. On the website, it displays all names for shared checklists, but puts the "author" of the specific checklist in bold, so I've set up the code to extract the name in bold.

@dbl3raf can you please confirm if surveyor sheets should be made for all grid cells or just the land ones?

**Comment by jeffreyhanson** [comment]
Here's a draft of a surveyor sheet so far. Although I haven't got the code working to insert the checklist data yet, @dbl3raf and @Louis-Backstromme can know if the first and last pages need tweaking?

[grid-10496.pdf](https://github.com/bird-team/brisbane-bird-atlas/files/2599120/grid-10496.pdf)

**Comment by jeffreyhanson** [comment]
Just tweaked it a bit, see this one:

[grid-10496.pdf](https://github.com/bird-team/brisbane-bird-atlas/files/2599159/grid-10496.pdf)

**Comment by dbl3raf** [comment]
Far out @jeffreyhanson this is brilliant! I really appreciate how much time and effort you are putting into this. A few tweaks / clarifications:

1. Remove space in the header "Archerfield SouthGrid"
2. To be clear, the date in the header refers to the date of the dataset, not necessarily the current date at the time the sheet is prepared. Might need to be a tweakable parameter when building the sheets.
3. in the "Total km" field in the table, just do km accurate to 1 decimal place
4. Would be great if we had the ability to edit the material on the last page - I can imagine us needing to improve and expand what's written here in due course.
5. Confirming that surveyor sheets need only to be made for land squares.

**Comment by dbl3raf** [comment]
@jeffreyhanson could you remind me how land and sea squares are differentiated from each other? Is there a flag somewhere, or do you just do it dynamically via overlay?

**Comment by jeffreyhanson** [comment]
Ok - can do. With regards to:

2. We could take the month/year for the most recent record in the database to automate that? 
4. Yeah, I agree. I have the work for this in the `surveyor-sheets` branch at the moment, and the surveyor sheets are created from a template rmarkdown document (https://github.com/bird-team/brisbane-bird-atlas/blob/surveyor-sheets/templates/surveyor-sheet.Rmd). If you needed to edit the text that appears on the last page you could edit this document and then remake the surveyor sheets, but please don't try editing it till I merged with the `master` branch. I'll add info about this to the wiki once the surveyor sheet functionality is added in.

**Comment by jeffreyhanson** [comment]
Sorry I forgot to mention, the land/marine grid cells are determined automatically using the land shapefile (see  https://github.com/bird-team/brisbane-bird-atlas/tree/surveyor-sheets/data/land)

**Comment by dbl3raf** [comment]
Great idea to take most recent date from the records and use that.

Excellent the "standard" text on the sheet will be editable once this is built.

at some point we might also need to expand the land boundary, e.g. to ensure we capture the far edge of tidal flats / port reclamation etc. So it's good to know that we could presumably sub in a new shapefile at https://github.com/bird-team/brisbane-bird-atlas/tree/surveyor-sheets/data/land to achieve this.

**Comment by Louis-Backstrom** [comment]
This all looks fantastic - great work @jeffreyhanson ! Don't have anything else to add at the moment - it'll be great to see what it looks like once populated.

**Comment by jeffreyhanson** [comment]
Ok - here's another version of the surveyor sheet with the checklist data. Can you let me know if it needs any tweaks? Also, I noticed that the checklist in the docx example only contains ~190 of the ~380 species in the `atlas_list.xlsx` file, so I've added a column to the `atlas_list.xlsx` specifying which species should be included in the surveyor sheets (note this is only on the `surveyor-sheets` branch at the moment) if you need to change which species should be displayed in the future.

[grid-10496.pdf](https://github.com/bird-team/brisbane-bird-atlas/files/2604119/grid-10496.pdf)

**Comment by dbl3raf** [comment]
Brilliant @jeffreyhanson ! The survey sheets should contain all 383 species (both the recent docx files contained 383), so it would definitely be good to expand the grid-10496.pdf example to include all 383 for now. Can you fit the whole lot onto 2 middle pages?

One remaining bug is the total species count - it's too high at the moment.

Otherwise I think we're good to go, unless @Louis-Backstrom can spot anything else?

Huge thanks as ever @jeffreyhanson

**Comment by Louis-Backstrom** [comment]
Looks great to me. Only thing I do wonder is if we can wrap the text for some of the longer location names onto 2+ lines? It might help it look a bit tidier, but I'm not sure if the system you've got in place currently allows this.

**Comment by dbl3raf** [comment]
Wrapping would be good if possible. Where the long names are hotspots, or personal locations from a "friendly" birder, we could probably get them shortened in the eBird system. Any time I see a hotspot with unnecessary words in it, I'm going in and editing it down within eBird.

**Comment by jeffreyhanson** [comment]
Yeah that's doable - perhaps we could set a maximum character limit per line and then break up location names via spaces such that each line contains complete words where the number of characters does not exceed the limit?

**Comment by jeffreyhanson** [comment]
This of course assumes that no one puts a location name in using camel or snake case (e.g. `BestBirdingHotspotInTheWorld` or  `Best_birding_hotspot_in_the_world`).

**Comment by dbl3raf** [comment]
if they do, they deserve to have it disappear!

**Comment by jeffreyhanson** [comment]
Ok I've fixed the problem with the total species count and added all the species to the checklist (wrapped across two pages). Can you let me know what you think? I haven't fixed the labels on the map yet - but it's on my todo list.

[grid-10496.pdf](https://github.com/bird-team/brisbane-bird-atlas/files/2612017/grid-10496.pdf)

**Comment by dbl3raf** [comment]
looks amazing @jeffreyhanson ! As we add more species to the Brisbane checklist, we might end up with pagination problems, but we can deal with that later. 

The only other remaining issue aside from the map labels was making the date in the header equal the month of the latest checklist, rather than the system date at the time of production of the survey sheet.

**Comment by Louis-Backstrom** [comment]
Looks fantastic! As Rich said, we might run into problems with page wrapping later as the list grows but that should be easy to address.

**Comment by jeffreyhanson** [comment]
Ok I've fixed the month/year issue - what character limit would you like to impose for the wrapping?

**Comment by dbl3raf** [comment]
I think about 25 characters should do it for the wrapping

**Comment by jeffreyhanson** [comment]
Ok can do

**Comment by jeffreyhanson** [comment]
Some grid cells have less than five checklists, what should rows in the leaderboard look like when there is no one in that position (e.g. a grid cell has 3 checklists by 3 different people, thus there are three people in the leaderboard - what should the remaining 2 rows look like?). At the moment I'm just leaving these rows blank, but let me know if you want some inspirational text or something (e.g. "Your name here!")?

**Comment by jeffreyhanson** [comment]
@dbl3raf and @Louis-Backstrom, I'm almost ready to begin merging the survey sheet branch with the master branch. If you try pushing things to the master branch while I'm attempting to merge the survey sheet branch, then this may result in some of your updates getting lost or some of my updates getting lost. Although it would be tricky---and not impossible---to track down "lost" updates, it would be preferable to avoid this.

So, could you please push any remaining local changes from your computers to the master branch? And then once this is done, could you please refrain from making any changes to the atlas (either via the GitHub interactive interface or locally on your computer) until I've given the OK that the merge has completed successfully? Can you please confirm when you're ready for me to begin starting them merge process.

**Comment by Louis-Backstrom** [comment]
Just merged my current work - all good from me.

**Comment by jeffreyhanson** [comment]
Great - thanks @Louis-Backstrom. @dbl3raf, can you let me know when I can begin the merge process?

**Comment by dbl3raf** [comment]
great! go ahead - I have no outstanding changes

**Comment by jeffreyhanson** [comment]
Ok great - I'll get started on the merge.

**Comment by Louis-Backstrom** [comment]
Looks awesome!

**Comment by jeffreyhanson** [comment]
It looks like everything is going well so far - I'm going to try rebuilding all the assets and creating all the survey sheets now.

**Comment by jeffreyhanson** [comment]
Ok, I've rebuilt the assets---including all 500+ surveyor sheets which are now available as separate files in their own release; https://github.com/bird-team/brisbane-bird-atlas/releases/tag/v.0.0.2 ---and the website so that the grid cells in the interactive maps now have links to their surveyor sheets.

For example, if you visit the magpie goose page (https://brisbanebirds.com/magpie-goose-anseranas-semipalmata.html#magpie-goose-anseranas-semipalmata), scroll down to the interactive map, click on the"Grid" checkbox, and then click on one of the grid cells, a popup should appear over the grid cell with the name of the grid cell. If you click on the name of the grid cell (in blue), your computer should start downloading the surveyor sheet for that grid cell (or you should be prompted to ask where you want to download it depending on your default browser settings).

@dbl3raf and @Louis-Backstrom, can you let me know if there are any further changes that need to be made to the surveyor sheets or their integration into the atlas?

Also, thank you very much waiting for me to complete the branch merge and double check everything's working properly post-merge. It looks everything is working, so please feel free to recommence updating the atlas.

**Comment by jeffreyhanson** [comment]
Since all the functionality to make the surveyor sheet maps is now in the main atlas repository (i.e. this repository), I've added a deprecation notice to the [sampling-grid-maps repository](https://github.com/bird-team/sampling-grid-maps).

**Comment by dbl3raf** [comment]
@jeffreyhanson - this is nothing short of miraculous, brilliant job. This fills in the last major gap in the functionality of the Atlas - we now have everything in place for the "launch" to go ahead on 1 Jan, when we introduce the website and how to contribute to the broader birding community. Once the instructions are up on the wiki for how to update the survey sheets, we can have a go at writing the grid square accounts to replace the Lorem ipsum on the survey sheets. A thousand thanks @jeffreyhanson

**Comment by jeffreyhanson** [comment]
No worries - ok great - I'll get started on updating the documentation then.

**Comment by Louis-Backstrom** [comment]
![image](https://user-images.githubusercontent.com/44354406/49070728-1d843100-f278-11e8-9cae-b8dc87738dfb.png)
Surveyor sheets are looking good @jeffreyhanson - just the clipping doesn't look like it's doing 25 characters? It seems to have done every single word for a new line which looks pretty messy. This is downloaded just now - is this intended?

**Comment by Louis-Backstrom** [comment]
Another possible bug - total species counts still look wrong. Looking at the Weller's Hill square, it says there are 47 species recorded but then on the table there are about 80. Not sure why this would be - could it be that it's only counting species recorded on complete checklists and Weller's Hill has a large number of incidentally-only recorded species or something?

**Comment by Louis-Backstrom** [comment]
![image](https://user-images.githubusercontent.com/44354406/49072267-9b960700-f27b-11e8-8a7c-e82b3ad37438.png)
see colin reid especially - 67 species (so more than total) but all incomplete checklists

**Comment by jeffreyhanson** [comment]
Yeah it looks like the wrapping is still not working correctly - thanks for catching that.

Yeah, the "summary table" only contains complete checklists but the "checklist table that lists each species separately" is based on all records (including incidental records). I agree this could be confusing. I could change it so that both tables use only complete checklists - what do you think?

**Comment by Louis-Backstrom** [comment]
I think it's important to have information on all the species recorded, incidental or not so I'd more err on the side of counting all species in all places. Perhaps we could meet in the middle and put like the "total checklist" species count how it currently is followed by the number inclusive of non-complete checklists in brackets? So for the total it would look like "47 (80)" and similar for each season. Perhaps you could also put it down in the species by species table with each species that has only been recorded incidentally in that particular season being represented by a small x rather than a large one? Species that have been reported both incidentally and not would still be big X.

Does that make sense? Thoughts @dbl3raf ?

**Comment by jeffreyhanson** [comment]
Hmm, I think if we included the total number in parenthesis we would need to rename the columns. I worry that numbers under a column named "checklists" that correspond to incidental records could be confusing (even if they are in parentheses).

**Comment by Louis-Backstrom** [comment]
Yeah, I do worry things could get confusing too. It's a tricky situation - not sure.

**Comment by jeffreyhanson** [comment]
Just a random thought - what is the main aim of the checklist table showing each different species? Is it meant to provide a quick reference for species which might potentially be in the grid cell to help with ids in the field? If so then maybe the the extra information about if species has been recorded in a checklist (big X) vs only in incidental records (small X) is unnecessary? Or is the main aim to provide a detailed understanding about how well the grid cell has previously been sampled? In which case the extra information is really important? Or is the main aim something else?

**Comment by jeffreyhanson** [comment]
I just fixed the bug with the label wrapping so I'll set up my computer to remake the assets over night.

**Comment by dbl3raf** [comment]
Well spotted on these issues, good pickup.

I think the best solution for the species count in the checklists section of the table is to remove the column labelled "Species" altogether. Thinking about it, we don't really need a species count there - the left hand part of the table is about _effort_ and so a species total is irrelevant really. Deleting this column will make the two sides of the table distinct and with clear purpose - the left hand side shows progress towards the _survey effort_ goals, and the right hand side is for some light hearted competition aimed at _stimulating further searches_ of the grid square.

Likewise, although I certainly see the point of the big X little x idea for the main checklist table, I think it introduces unnecessary complexity and won't really drive any particular desirable behaviour from surveyors. The main purpose of that table is to show what species have been recorded from the grid cell in different seasons. I don't think it really matters if they are from complete or incidental checklists.

The only omission then is that we won't be showing the total species count for the grid square, which is kind of a nice statistic to have somewhere. Birders will be keen to know it.

So my final suggestion is to have another row under the five observers called **Total**, which would have the complete species count (whether from complete or incomplete checklists), total number of complete checklists, and total number of incomplete checklists. This would help round everything off, and contain all the information we need I think.

**Comment by jeffreyhanson** [comment]
Ok, sounds good. Could you please edit the mockup docx with the new summary table specification so I know exactly what the table should look like? The only potential issue I can think of would be that having the total species counts under the leader board might lead people to conclude that this is the total amount of species observed by the leader board and the total amount of species observed in the grid cell? What if we just added the species count to the header or footer?

**Comment by jeffreyhanson** [comment]
The updated surveyor sheets are being pushed now, can you please confirm if the 25 character wrap is good enough (e.g. see here for an example https://github.com/bird-team/brisbane-bird-atlas/releases/download/v.0.0.2/grid-10097.pdf)?

**Comment by dbl3raf** [comment]
that preview with the new wrapping looks good to me. I'm going to change all of the Sushinsky study sites to much shorter names - they somewhat dominate every grid square at the moment!

**Comment by dbl3raf** [comment]
@jeffreyhanson - I agree, we could just add the total species count to the header

**Comment by jeffreyhanson** [comment]
Ok great - just to confirm, how should the header be updated? E.g. "Atlas of the Birds of Brisbane
Container Terminal Grid Square, Oct 2018 (30 species recorded)"?

Yeah, I've noticed that a lot of eBird sampling localities include the phrase `... at (LATITUDE, LONGITUDE)` (e.g. `Sushinsky Study Point count 578 at 27.37, 153.178`).  If you think it's a good idea, I could update the label processing to omit the coordinates (e.g. show  `Sushinsky Study Point count 578` instead)?

**Comment by jeffreyhanson** [comment]
But this could be confusing for people that want to use the atlas and eBird in combination, because if we tried "cleaning" the eBird locality names then the names in the altas would differ from those in the eBird dataset. Is this important? Do you think it's important to maintain consistency with eBird data set? Or are you suggesting that we identify such verbosely named sites and update their names in the primary eBird database? If so, I could send you a list of sites that all contain coordinates in their names.

**Comment by Louis-Backstrom** [comment]
I think it's important to have consistency between the two so wouldn't advise changing the names in our dataset. Changing names in the eBird dataset is a possibility but a pretty big task as there are a lot of personal locations with co-ords as that's how eBird default creates them. Plus, I'm not aware of there being an easy way to batch edit names from the backend either.

Two other options I see being possible - 1. decreasing the font size - the font isn't too big at the moment but I reckon it would be fine a size or two smaller, and that would help clear clutter a bit 2. semi-truncating super long (e.g. over 50 characters) names with `...` so the user knows that the name is still the same but we're just hiding the end of it because it's so long.

Thoughts?

**Comment by jeffreyhanson** [comment]
Ok I see - I didn't know that was the default naming scheme, so yeah updating them all sounds like a pretty onerous task.

Yeah we could simply make the font smaller. I haven't actually tried printing out the surveyor sheets so I don't know how big they appear on the printed page? But the locality name text looks slightly bigger than the loreum ipsum in the surveyor sheets, so we could try decreasing the locality font size till it's about equal?

**Comment by dbl3raf** [comment]
yep sounds good - reducing the font size a little in the labels will be fine. We have full control within the eBird system of all hotspot names, and I can edit long ones down as we find them, but as @Louis-Backstrom says, personal locations are owned by their creator and can only be changed by them. for particularly extreme cases I can contact observers and ask them to change the names, but this is a relatively low priority task for sometime in the distant future. For now, I think we just faithfully reproduce the names as they appear in eBird, and just live with the fact that some grid square maps are cluttered. In one sense it doesn't really matter - a very cluttered grid square is almost certainly going to be a low priority for the _Atlas_ because it is probably going to be well-surveyed.

**Comment by dbl3raf** [comment]
And yes "Atlas of the Birds of Brisbane Container Terminal Grid Square, Oct 2018 (30 species recorded)" sounds great, thanks @jeffreyhanson

**Comment by Louis-Backstrom** [comment]
Just noticed that the maps don't show the border where the border lies somewhere along a square - I think this is probably a good thing to have as many folks won't know the specific ins and outs of each section of the border and we don't want them submitting stuff in Moreton Bay etc. (not that we're greedy or anything!)

**Comment by Louis-Backstrom** [comment]
![image](https://user-images.githubusercontent.com/44354406/49139381-62be6680-f33d-11e8-926f-0fe93c2e5115.png)
Also, this is a disaster. I can't think of a much better solution to the cramming problem at the moment but I think we need to improve it somehow...

Finally - are we able to make it such that the filename for downloaded survey sheet pdfs follows the format `grid-[grid number]-[grid name]` or similar just so it's easier to manage if you have multiple sheets downloaded? Makes it much easier to recognise.

**Comment by Louis-Backstrom** [comment]
![image](https://user-images.githubusercontent.com/44354406/49139560-d9f3fa80-f33d-11e8-812d-2518f97f2c67.png)
OCC is even worse... But as Rich says, it seems to be exclusively a problem for squares that have enough data, not squares that have too little. We'll see.

**Comment by jeffreyhanson** [comment]
Sorry, I don't I understand your comment about the maps not showing the border. Do you mean the Brisbane LGA border?

Yeah, I agree - that's massive problem for grid cells with lots of sampling localities. What if we only showed labels for the top 10 sampling localities with the most number of checklists in each grid cell?

I'm remaking the assets now with smaller labelling text - but I don't think this will fix the maps for grid cells with lots of sampling localities.

**Comment by jeffreyhanson** [comment]
Hmm, we could have the file names show the grid name in addition to the id number. I'll see what I can do.

**Comment by Louis-Backstrom** [comment]
> Sorry, I don't I understand your comment about the maps not showing the border. Do you mean the Brisbane LGA border?
> 

Yeah - obviously a lot of squares have the LGA border running through them and it's not currently displayed on those squares but I think it would be beneficial if it did.

Something like that would be good for sure - I don't think there's one perfect solution to the overcramming problem unfortunately.

Thanks for all this!

**Comment by jeffreyhanson** [comment]
Ah ok - yeah - we could include the border in the maps - but I worry this might clutter the maps even more? How important is it that surveyors make sure that they do checklists inside the Brisbane LGA on border grid cells?

**Comment by dbl3raf** [comment]
I think we should definitely include the border again (it used to be in there right?). It's just informative to know where it is - surveys outside the Brisbane border are not included in the eBird download anyway, and if we want them to be, we can buffer future downloads (this process would occur outside the _Atlas_ anyway, so there's nothing to change in the _Atlas_ code to deal with this).

And on the clutter issue, I appreciate the thoughtful discussion. On balance, I think we should do precisely nothing - just leave as it is, albeit with the further label font size reduction, which will help. There will only be a few cluttered squares, and by definition they will be well visited and not of huge strategic importance for the _Atlas_.

For the record (in case we ever reopen this giant issue!), other options include:
- only including hotspots
- truncating the names
- only including, e.g. the top ten sites by species richness

**Comment by jeffreyhanson** [comment]
Ok - with regards to the the LGA border:
1. what color should the border be?
2. what line width should be used? E.g. same width as grid cell lines? or thicker?
3. should the LGA border be placed underneath the grid cell lines or on top of the grid cell lines?

Also, I've just started pushing the new surveyor sheets with smaller text (the original was size 2.5, and this is with size 2.0), can you let me know if this is a good size or if it should be smaller?

**Comment by jeffreyhanson** [comment]
Looks like I made a mistake when editing the code to add grid names to the surveyor sheet pdf files, so I'll need to remake them again. Just as a heads up, the release v.0.0.2 with the surveyor sheets is now empty. I've attached an example pdf though (with font size 2.0 -- the font sizes are different from ms word), can you let me know what you think of the font size in the labels?

[grid-10017.pdf](https://github.com/bird-team/brisbane-bird-atlas/files/2624395/grid-10017.pdf)

**Comment by Louis-Backstrom** [comment]
> Ok - with regards to the the LGA border:
> 
> 1. what color should the border be?
> 2. what line width should be used? E.g. same width as grid cell lines? or thicker?
> 3. should the LGA border be placed underneath the grid cell lines or on top of the grid cell lines?
> 
> Also, I've just started pushing the new surveyor sheets with smaller text (the original was size 2.5, and this is with size 2.0), can you let me know if this is a good size or if it should be smaller?

My thoughts - 1. Maybe a black line with white edges? 2. Pretty thick, I'd go a bit thicker than the cell lines but not too much. 3. Underneath.

Font size looks good now, I don't think there's much benefit in going smaller as it'll get too hard to read.

**Comment by jeffreyhanson** [comment]
Yeah sounds good - I don't think I can do a line with a "fill" and a "border colour" though - so the line will have to be a single colour. How would black work? Though it might be easier to see on the imagery if its a light colour? Perhaps a light blue? If you give me a hex code (e.g. `#33F7FF`), I can specify the colour exactly (see here for a colour picker: https://htmlcolorcodes.com/color-picker/)

**Comment by dbl3raf** [comment]
@jeffreyhanson that example map looks great (just to confirm I assume you knew the numbers in the table are off, which is why you're re-running it).

Maybe grey for the LGA border (something like `#cfcfcf`) so that it has the same effect as the light blue, but isn't too garish?

Thanks for your patience @jeffreyhanson in making sure we get this right - it's a critically important part of the _Atlas_.

**Comment by jeffreyhanson** [comment]
@dbl3raf, Ok - can do. I wasn't aware of any problems with the numbers in the table - could you please expand on this so I can fix it?

**Comment by jeffreyhanson** [comment]
Here's an example grid map with the study area (i.e. Brisbane LGA lines) with that color, can you let me know what you think?

![test](https://user-images.githubusercontent.com/3610005/49189318-8a531480-f3b9-11e8-8f21-f3df62bab8f8.png)

**Comment by jeffreyhanson** [comment]
Hmm, it looks like the code simplified this data set to reduce the size and loading times for the interactive widgets in the html version, I'll remake the image using a non-simplified LGA boundary.

**Comment by jeffreyhanson** [comment]
It looks like the underlying study area data with the LGA boundary (i.e. https://github.com/bird-team/brisbane-bird-atlas/tree/master/data/study-area) is still pretty coarse (see image below which uses the data in the repo without simplifying it) . I guess we could add the raw LGA boundary data to the repo too and use that for the surveyor sheets. What do you think?

![test2](https://user-images.githubusercontent.com/3610005/49190813-ac4f9580-f3bf-11e8-8272-9cf4c2da5a28.png)

**Comment by dbl3raf** [comment]
the numbers in the table seem strange because the lead person has only 1 complete checklist, despite the fact there have been 239 complete checklists during the _Atlas_ period. Stephen Murray has made at least 9 complete checklists for the Boondall Wetlands Reserve hotspot, and has seen 109 species there - https://ebird.org/australia/hotspot/L921912?yr=all&m=&rank=mrec&sortBy=cl

**Comment by dbl3raf** [comment]
Would be great to get a less coarse border, even for use in the calculations - could the coarse border mean that some of the coastal eBird locations are being missed out of the analysis?

**Comment by dbl3raf** [comment]
Shall we try going darker for the border colour - maybe `#838282`

**Comment by jeffreyhanson** [comment]
Ok - thanks for explaining that - I'll have a look and see what I can do. I'll change the color and post another image to see what you think. Yeah, the coarse border could mean that some coastal eBird locations are omitted, but using a less coarse border will increase the time it takes to build the assets. If you would like to prepare a new study area shapefile, I can have a go at rebuilding the assets with that and see how it goes?

**Comment by jeffreyhanson** [comment]
![test3](https://user-images.githubusercontent.com/3610005/49323382-5ff28a00-f566-11e8-9859-ab178ad80298.png)

**Comment by jeffreyhanson** [comment]
Ok I think I fixed the bug causing incorrect leader board data, here's the updated surveyor sheet:

[grid-10017-shorncliffe.pdf](https://github.com/bird-team/brisbane-bird-atlas/files/2635565/grid-10017-shorncliffe.pdf)

**Comment by Louis-Backstrom** [comment]
Looks better - still surprised that Steve Murray has so many incomplete checklists but I guess it's plausible. Haven't looked at the data.

The border still looks really wonky - even with the coarse version I wouldn't expect it to overlap like it has? 

![image](https://user-images.githubusercontent.com/44354406/49324037-95e93b80-f571-11e8-8a25-e3e4df84c211.png)
 The highlighted bits in particular seem strange?

![image](https://user-images.githubusercontent.com/44354406/49324097-f9279d80-f572-11e8-9729-76a358b2afce.png)
Thisis what it looks like on the widget - not very precise but at least it's accurate - so something seems to be happening during the production of the survey sheets that's getting it all wonky.

**Comment by jeffreyhanson** [comment]
Yeah, that's odd - not sure why it's any different. They're both made from the same underlying shapefile data.

**Comment by jeffreyhanson** [comment]
Well, the surveyor sheet map shows both the extent of the "land" and "marine" areas, and that interactive widget only seems to show the "land" extent (as specified in the `atlas_list.xlsx` file). Maybe it would be worth comparing for a species that is associated with both "land" and "marine" areas (e.g. https://brisbanebirds.com/arctic-jaeger-stercorarius-parasiticus.html#arctic-jaeger-stercorarius-parasiticus)?

**Comment by Louis-Backstrom** [comment]
![image](https://user-images.githubusercontent.com/44354406/49334215-2a0fdd00-f61c-11e8-9f7e-7df3a67dbbce.png)
 it looks fine even for Arctic Jaeger - so I'm not sure what is causing the issue.

**Comment by jeffreyhanson** [comment]
Hmm, that's odd. @dbl3raf and @Louis-Backstrom, can you please let me know if you think the resolution/coarseness of the boundaries in the surveyor sheets is acceptable or not? If the coarseness is acceptable, I can work on fixing the yellow lines that @Louis-Backstrom has identified in the image. If the coarseness is not acceptable, then we need to talk about or generate a new shapefile to delineate the LGA/coastline boundary in the surveyor sheet (and so there's not need for me to bother working out what's happening with the yellow lines just yet).

**Comment by dbl3raf** [comment]
The coarseness is definitely a problem - maybe we could try with a finer map? If it slows the build too much, we can figure out how to solve it?
[QSC_Extracted_Data_20181202_120519485000-14884.zip](https://github.com/bird-team/brisbane-bird-atlas/files/2636526/QSC_Extracted_Data_20181202_120519485000-14884.zip)

**Comment by dbl3raf** [comment]
the layer I've attached has a much better boundary for our use - it extends much further into the intertidal - perhaps it is a 2018 update? It can be explored interactively [here](https://results.ecq.qld.gov.au/local_area_maps/).

**Comment by jeffreyhanson** [comment]
Hmm, that's not quite going to work because we also need the "marine" extent too. I've written some code (https://github.com/bird-team/study-area-data) to use the Australia-wide LGA data to create the Brisbane LGA extent and the marine extent (using 200 nautical mile buffer). I'll see if I can find an updated version of the Australian LGA data and use that.

**Comment by dbl3raf** [comment]
> 
> Looks better - still surprised that Steve Murray has so many incomplete checklists but I guess it's plausible. Haven't looked at the data.
> 
This is now correct - the incomplete lists are from the Eremaea data, which was all imported into eBird as an Incidental checklist type, so therefore treated incomplete. This is a major problem with the historical data, since almost all of the Eremaea lists actually represented complete data. We might be able to convince some of the more active observers to change their old lists, but that's a project for much, much later on!

**Comment by jeffreyhanson** [comment]
Ok, here's an updated version of the study area shapefile data (with "land" and "marine" areas). @dbl3raf can you take a look at it?

[exports.zip](https://github.com/bird-team/brisbane-bird-atlas/files/2636591/exports.zip)

**Comment by jeffreyhanson** [comment]
It looks like this study area shapefile doesn't overlap with a few of the terrestrial and marine grid cells (i.e. in the `data/grid/grid.shp` file. If that shapefile is acceptable, can you also please confirm if it's acceptable to delete these grid cells? It's critical that the study area shapefile overlaps with all the grid cells.

**Comment by dbl3raf** [comment]
I'm happy for those deletions - I did notice that the study area shapefile has a few holes in it around Moreton Island - showing through as white areas in the attached screengrab
![image](https://user-images.githubusercontent.com/23551540/49338802-ed230500-f672-11e8-91f7-9c5d52c401d5.png)

**Comment by jeffreyhanson** [comment]
Ok yeah, those gaps are a bit odd. Maybe they're due to the geometry simplification processing?

Also, I've been thinking a bit about this and I'm a bit hesitant to play around with the underlying study area and grid cell shapefiles. I worry that changing these datasets might introduce errors somewhere else into the atlas. So, I had a go at using the original LGA data to create the study area shapefile again, but this time I removed all the code used to simplify the geometries to retain as much detail as possible. I've also edited the atlas code to avoid any simplification processing too. I've attached a grid map that shows this data, can you let me know if this is good enough? If it is good enough, that will make things a lot easier on my end.

![map](https://user-images.githubusercontent.com/3610005/49339071-ca471f80-f677-11e8-8f28-b36ecd306a3e.png)

**Comment by dbl3raf** [comment]
sure @jeffreyhanson - that looks totally great! I'm happy to proceed on this basis

**Comment by jeffreyhanson** [comment]
Ok - I'll push the fixes to the repo and start remaking all the assets + surveyor sheets

**Comment by Louis-Backstrom** [comment]
New surveyor sheets look great, but some squares are coming up with "not found" when I try to download them.

Lawton, Northbrook Mountain, Ted Smout etc. - seems to be some of the border ones? Plenty of others are working fine.

**Comment by jeffreyhanson** [comment]
Yeah - it looks like GitHub is sometimes randomly failing to accept the new surveyor sheets when the code tries to upload them. I'm updating the code to be more resilient to these random failures.

**Comment by Louis-Backstrom** [comment]
OK great - that's a bit weird behaviour.

Once that is fixed I think everything with the surveyor sheets will be in place for launch.

**Comment by Louis-Backstrom** [comment]
![image](https://user-images.githubusercontent.com/44354406/49410731-d33c0c00-f7b1-11e8-8457-14b212988df3.png)
@jeffreyhanson the season ordering is wrong by the looks of it - should go Su/Au/Wi/Sp correct?

**Comment by jeffreyhanson** [comment]
Yeah - you're right - thanks for catching that.

**Comment by Louis-Backstrom** [comment]
Looks like all the surveyor sheets are displaying correctly and the seasons are in the right order (although the last few builds seem to have failed). Are we happy to close this issue?

**Comment by jeffreyhanson** [comment]
Yeah, I think so. @dbl3raf, any further changes required?

The build failure was caused by a missing quote (`"`) symbol in one of the figure captions - I've just added the missing quote so hopefully that fixes the build.

**Comment by Louis-Backstrom** [comment]
Not from me at the moment - we can finally close this behemoth of an issue! Further changes might pop up in the future, but we can create new issues for them I think.

**Comment by dbl3raf** [comment]
Brilliant job @jeffreyhanson - there's one final isssue, spotted by @Louis-Backstrom. It seems that the total species count for each observer in the table is calculated only on complete checklists. If easy to alter, this should be based on all lists, whether complete or incomplete. If not easy to alter, no need to worry about it.

**Comment by jeffreyhanson** [comment]
No worries - that's an easy fix (just deleting 15 characters of code). I'll push it in a moment and begin remaking the assets.

**Comment by jeffreyhanson** [comment]
Ok - I've just pushed the assets with the fixed leader board species counts - so I'll close this issue.

**Comment by Louis-Backstrom** [comment]
I'm gonna reopen this just because it seems there's only surveyor sheets for the land cells that are named. When I was naming them all I didn't name a few on the foreshore areas esp. along the coast as I didn't realise they were classified as land and not marine - so there are a few that are still called "grid cell #####" but are land, not marine. Do we want surveyor sheets for these too? Not really a major issue - and probably alternatively fixed by adding some names to these squares.

**Comment by jeffreyhanson** [comment]
I would hazard a guess that these grid cells should be named and should also have surveyor sheets - but what do you think @dbl3raf? Yeah, this can indeed be fixed by updating the names for these grid cells in `data/grid/grid-metadata.xlsx` and then rebuilding the assets.

**Comment by Louis-Backstrom** [comment]
Right - I've just gone and added names for all the ones I can see that were missing. In theory, next time the assets are rebuilt this should go away? I think we can reclose.

**Comment by jeffreyhanson** [comment]
Ok - awesome - I've started rebuilding the assets - and I'll let you know when I've pushed them all to the release.
